gdb-lite/gdb-lite.c

```diff
@@ -21,9 +25,25 @@ int main(int argc, char *argv[]) {
 
   while (fgets(input, MAX_INPUT_LENGTH, stdin) != NULL) {
     // Interpret command here
-    if (strcmp(input, "run") == 0) {
-      int code = run(fname);
-      printf("%s exited with code %d\n", fname, code);
+    if (strcmp(input, "run\n") == 0) {
+
+        //If it is the child then we wanna run this
+        int code = run(fname);
+        printf("%s exited with code %d\n", fname, code);
+      
+      
+    } else if (strcmp(input, "break\n") == 0) {
+      pid = fork();
```

Comment:
We fork the process, the child becomes the one we breakpoint on and the parent is the debugger.